### PR TITLE
Added Masked behaviour

### PR DESCRIPTION
**Pull request by NiroDev**
Added SeeThrough shader
Compatibility for Mirage & ModelReplacementApi
Utils for Material replacing
Unity Audio folder
several bug fixes

### CODE REVIEW

File: LethalMon/Behaviours/MaskedTamedBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,1125 @@
...
+            lendMaskCooldown = GetCooldownWithId(CooldownId);
+
+            if (Mask != null)
+            {
+                originalMaskParent = Mask.transform.parent;
+                originalMaskLocalPosition = Mask.transform.localPosition;
+            }
+
+            if (ownerPlayer != null) // Baye Layers: Roam, Follow, Wait
+                Masked.creatureAnimator.Play("Base Layer.Wait"); // wip, not working yet
```
Comment (Feiryn) [inline LethalMon/Behaviours/MaskedTamedBehaviour.cs:396]: Is it working or not?

File: LethalMon/Behaviours/MaskedTamedBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,1125 @@
...
+        internal static List<Tuple<AudioClip, AudioClip>> GhostVoices = new List<Tuple<AudioClip, AudioClip>>();
+        internal float timeTillNextGhostVoice = 0f;
+
+        internal AudioSource? farAudio = null;
+        #endregion
+
+        #region Cooldowns
+        private static readonly string CooldownId = "masked_lendmask";
+
+        internal override Cooldown[] Cooldowns => new[] { new Cooldown(CooldownId, "Lending mask", 2f) };
```
Comment (Feiryn) [inline LethalMon/Behaviours/MaskedTamedBehaviour.cs:136]: Can you add a config field for the cooldown time and also for the view distance?

File: LethalMon/Behaviours/MaskedTamedBehaviour.cs (hunk 1 above)
Comment (NiroDev) [reply LethalMon/Behaviours/MaskedTamedBehaviour.cs:396]: not working yet

File: LethalMon/Behaviours/MaskedTamedBehaviour.cs (hunk 2 above)
Comment (NiroDev) [reply LethalMon/Behaviours/MaskedTamedBehaviour.cs]: Done

**Pull request merged**